api/pom.xml

```diff
@@ -3,7 +3,7 @@
     <parent>
         <groupId>org.openmrs.module</groupId>
         <artifactId>queue</artifactId>
-        <version>2.3.0-SNAPSHOT</version>
+        <version>2.4.0-SNAPSHOT</version>
```

Comment:
Why is this getting changed from 2.3.0-SNAPSHOT -> 2.4.0-SNAPSHOT?  This would seem to indicate that this branch has not been updated with the latest from main.  Please merge main up to this branch so these changes no longer appear as within this PR.  I don't think we need to explicitly change the version in this PR.

Reply:
The version bump is intentional. I made a backwards-incompatible change to the endpoint for transitions from `/queue-entry-transition` to `queue-entry/transition` (to namespace it better).

Reply:
Thanks @chibongho.  If we want to version bump, for Maven modules we can't just do a release via updated the version number... basically, we'd want to do a release OpenMRS CI (which would release 2.3.0 and then automatically bump the version to 2.4.0-SNAPSHOT) before merging this branch in.

But the other thing to check is that if there already are breaking changes in the 2.3.0-SNAPSHOT (ie since the 2.2.0 release) we won't need to do another release.

Reply:
Spoke to Mike. I did indeed mess up by local branch. main branch is already at 2.4 (and it hasn't been released yet) so i'll just keep that.